FV3GFS/FV3GFS_io.F90

```diff
@@ -4624,9 +4674,133 @@ subroutine gfdl_diag_register(Time, Sfcprop, Gfs_diag, Cldprop, &
     Diag(idx)%mod_name = 'gfs_phys'
     Diag(idx)%intpl_method = 'bilinear'
     allocate (Diag(idx)%data(nblks))
-    do nb = 1,nblks
-      Diag(idx)%data(nb)%var2 => Gfs_diag(nb)%uswsfci(:)
-    enddo
+    ! This diagnostic is always meant to refer to what the model felt, so it
+    ! refers to the override flux when overriding and the RRTMG flux when not.
+    if (override_surface_radiative_fluxes) then
+      do nb = 1,nblks
+        Diag(idx)%data(nb)%var2 => Gfs_diag(nb)%uswsfci_override(:)
+      enddo
+    else
+      do nb = 1,nblks
+        Diag(idx)%data(nb)%var2 => Gfs_diag(nb)%uswsfci(:)
+      enddo
+    endif
+
+    if (override_surface_radiative_fluxes) then
+      idx = idx + 1
+      Diag(idx)%axes = 2
+      Diag(idx)%name = 'DLWRF_from_rrtmg'
+      Diag(idx)%desc = 'Native RRTMG surface downward longwave flux'
+      Diag(idx)%unit = 'W/m**2'
+      Diag(idx)%mod_name = 'gfs_phys'
+      Diag(idx)%cnvfac = cn_one
+      Diag(idx)%time_avg = .TRUE.
+      Diag(idx)%intpl_method = 'bilinear'
+      Diag(idx)%coarse_graining_method = 'area_weighted'
+      allocate (Diag(idx)%data(nblks))
+      do nb = 1,nblks
+        Diag(idx)%data(nb)%var2 => Gfs_diag(nb)%dlwsfc(:)
+      enddo
```

Comment:
@lharris4 do you happen to know the difference between [this version](https://github.com/NOAA-GFDL/SHiELD_physics/blob/69d13c245348264f1bf12fd871261d5e25e36695/FV3GFS/FV3GFS_io.F90#L3112-L3124) of the interval-averaged surface downward longwave flux diagnostic and [this version](https://github.com/NOAA-GFDL/SHiELD_physics/blob/69d13c245348264f1bf12fd871261d5e25e36695/FV3GFS/FV3GFS_io.F90#L3938-L3950)?  Something similar can be asked for `ULWRFsfc` vs. `ULWRF`.  

It looks like in the PIRE simulations `DLWRFsfc` and `ULWRFsfc` were used.  Are we to assume those are the proper versions of these diagnostics, and that the `DLWRF` and `ULWRF` diagnostics are not to be used?  The reason for asking is that it influences how I would define these RRTMG fields — currently they are defined like the `DLWRF` and `ULWRF` versions, which were the only versions present in FV3GFS.

Reply:
Per our offline conversation I went ahead and used `DLWRFsfc` and `ULWRFsfc` as the source of truth for these RRTMG fields.

Reply:
Actually I am not entirely sure what the different diagnostics represent. It looks like `DLWRF` etc. use a [zenith-angle adjustment](https://github.com/NOAA-GFDL/SHiELD_physics/blob/69d13c245348264f1bf12fd871261d5e25e36695/GFS_layer/GFS_physics_driver.F90#L868-L884) whereas `DLWRFsfc` [is straight from RRTMG](https://github.com/NOAA-GFDL/SHiELD_physics/blob/69d13c245348264f1bf12fd871261d5e25e36695/gsmphys/radsw_main.f#L61-L88) and does not include this correction.

Reply:
@linjiongzhou @MingjingTong-NOAA do either of you know more about the difference between `DLWRF` and `DLWRFsfc`? This may explain why we see the four 'peaks' in our radiation diagnostics.

Reply:
That's an important point @lharris4, thanks.  In the case of the longwave fluxes, those are "surface temperature adjusted" and indeed those are the ones that we should actually use here since that is in fact what we are overriding (despite what we may have output in past simulations).  I can fix that tomorrow morning.

In the case of shortwave it looks like [there are instantaneous diagnostics for the zenith-angle-adjusted versions](https://github.com/NOAA-GFDL/SHiELD_physics/blob/69d13c245348264f1bf12fd871261d5e25e36695/FV3GFS/FV3GFS_io.F90#L4607-L4629) but not interval-averaged.  I can add those as well, as I think they could be useful going forward (those are again what we are actually overriding).

Reply:
I took a look at the code. It seems DLWRFsfc directly comes out of radiation calculation in subroutine GFS_radiation_driver, while DLWRF is time step adjusted sfc downward lw flux coming out from subroutine dcyc2t3.

!    dcyc2t3 fits radiative fluxes and heating rates from a coarse      !
!    radiation calc time interval into model's more frequent time steps.!

Reply:
Based on my understanding, Xsfc is the averaged radiation flux within the radiation time step (e.g., 1 hour), Xsfci is the averaged radiation flux within the physics time step (e.g., 150s), Xsfci is called instantaneous flux probably due to its short time interval. Xsfci is adjusted from Xsfc based on the difference of solar zenith angle between the radiation and physics time steps. The shape of Xsfci should be taller and narrower than the Xsfc. As for X, it is Xsfci * dt, which should hold a different unit of J/m^2. It is a time-accumulated variable.

Reply:
Per https://github.com/NOAA-GFDL/SHiELD_physics/pull/31#discussion_r1373903230, in 176da07 I went ahead and:

- Switched the toggling between RRTMG and overridden fluxes to occur only for the diagnostics that reflect the "adjusted" flux, i.e. `DLWRF`, `DLWRFI`, `DSWRF`, `DSWRFI`, `USWRF`, and `USWRFI`, since that is ultimately what we are overriding.  
    - Note that `DSWRF` and `USWRF` are newly added diagnostics that did not exist previously.
- I took the liberty to add more detail to the `desc` fields of all of the surface radiative flux diagnostics, to make clear the distinction between fields that come unadjusted from the radiative transfer code, and fields that are adjusted.

@lharris4 @MingjingTong-NOAA @linjiongzhou let me know if you agree with these changes.  

My read of the "instantaneous" description is that it is to distinguish from "interval-averaged" fields ("interval-average" meaning fields for which `Diag(idx)%time_avg = .TRUE.`).  The interval averages [are computed by dividing by the length of the averaging interval](https://github.com/NOAA-GFDL/SHiELD_physics/blob/69d13c245348264f1bf12fd871261d5e25e36695/FV3GFS/FV3GFS_io.F90#L7126), so the accumulated time-weighted fluxes of `Diag%dlwsfc` et al. are ultimately transformed back to having units of W/m^2.

Reply:
@spencerkclark, you are right. They are eventually transformed back to the unit of W/m^2 in FV3GFS_io.F90, for both Xsfc and X. Thanks for digging into this and clarifying it.